test/bayesian_linear_regression.jl

```diff
@@ -121,4 +121,24 @@ end
             @test cov(f′(X′, Σy)) ≈ cov(f′2(X′, Σy))
         end
     end
+    @testset "sampling functions" begin
+        rng, N, N_, D, samples = MersenneTwister(123456), 11, 5, 3, 10_000_000
+        X, f, Σy = generate_toy_problem(rng, N, D)
+
+        g = rand(rng, f)
+        @test g(X) == g(X)  # check the sample doesn't change between evaluations
+
+        Xc = ColVecs(X)
+        Xr = RowVecs(X')
+        @test g(X) == g(Xc)
+        @test g(X) == g(Xr)
+
+        # test statistical properties of the sampled functions
+        gs = rand(rng, f, samples)
+        Y = hcat(map(h -> h(X), gs)...)
```

Comment:
```suggestion
        Y = reduce(hcat, map(h -> h(X), gs))
```
generally much kinder on the compiler because splatting is only really designed for a few arguments, or something like that. Either way, there's a specialised method of `reduce` for both `hcat` and `vcat` which are generally a good choice if the number of things you're concatenating is really large.

Reply:
thanks - I did think that was a horrendous way of doing it :D